core/src/main/java/org/mapfish/print/processor/jasper/LegendProcessor.java

```diff
@@ -377,54 +377,59 @@ private IconTask(
     public Object[] call() throws Exception {
       return context.mdcContextEx(
           () -> {
-            BufferedImage image = null;
             final URI uri = this.icon.toURI();
-            final String metricName =
-                LegendProcessor.class.getName() + ".read." + StatsUtils.quotePart(uri.getHost());
-            try {
-              if (this.icon.getProtocol().equals("data")) {
-                image = ImageIO.read(this.icon);
-              } else {
-                this.context.stopIfCanceled();
-                final ClientHttpRequest request =
-                    this.clientHttpRequestFactory.createRequest(uri, HttpMethod.GET);
-                final Timer.Context timer =
-                    LegendProcessor.this.metricRegistry.timer(metricName).time();
-                try (ClientHttpResponse httpResponse = request.execute()) {
-                  if (httpResponse.getStatusCode() == HttpStatus.OK) {
-                    image = ImageIO.read(httpResponse.getBody());
-                    if (image == null) {
-                      LOGGER.warn(
-                          "The URL: {} is NOT an image format that can be decoded", this.icon);
-                    } else {
-                      timer.stop();
-                    }
-                  } else {
-                    LOGGER.warn(
-                        "Failed to load image from: {} due to server side error.\n"
-                            + "\tResponse Code: {}\n"
-                            + "\tResponse Text: {}\n"
-                            + "\tWith Headers:\n\t{}",
-                        this.icon,
-                        httpResponse.getStatusCode(),
-                        httpResponse.getStatusText(),
-                        String.join(
-                            "\n\t", Utils.getPrintableHeadersList(httpResponse.getHeaders())));
-                  }
+            final BufferedImage image = loadImage(uri);
+            final URI report = createSubReport(image, this.iconDPI, this.tempTaskDirectory);
+            return new Object[] {null, image, report.toString(), this.level};
+          });
+    }
+
+    private BufferedImage loadImage(final URI uri) {
+      final String metricName =
+          LegendProcessor.class.getName() + ".read." + StatsUtils.quotePart(uri.getHost());
+      BufferedImage image = null;
+      try {
+        if (this.icon.getProtocol().equals("data")) {
+          image = ImageIO.read(this.icon);
+        } else {
+          this.context.stopIfCanceled();
+          final ClientHttpRequest request =
+              this.clientHttpRequestFactory.createRequest(uri, HttpMethod.GET);
+          try (Timer.Context timer = LegendProcessor.this.metricRegistry.timer(metricName).time()) {
```

Comment:
It can't simply be?
```suggestion
          try (LegendProcessor.this.metricRegistry.timer(metricName).time()) {
```

Reply:
This timer must be declared in the try. I will rename it to ignored.